fwprovider/acme/resource_acme_plugin.go

```diff
@@ -0,0 +1,326 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package acme
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64default"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/bpg/terraform-provider-proxmox/proxmox"
+	"github.com/bpg/terraform-provider-proxmox/proxmox/cluster/acme/plugins"
+)
+
+var (
+	_ resource.Resource                = &acmePluginResource{}
+	_ resource.ResourceWithConfigure   = &acmePluginResource{}
+	_ resource.ResourceWithImportState = &acmePluginResource{}
+)
+
+// NewACMEPluginResource creates a new resource for managing ACME plugins.
+func NewACMEPluginResource() resource.Resource {
+	return &acmePluginResource{}
+}
+
+// acmePluginResource contains the resource's internal data.
+type acmePluginResource struct {
+	// The ACME account API client
+	client plugins.Client
+}
+
+// acmePluginCreateModel maps the schema data for an ACME plugin.
+type acmePluginCreateModel struct {
+	// API plugin name
+	API types.String `tfsdk:"api"`
+	// DNS plugin data
+	Data types.Map `tfsdk:"data"`
+	// A list of settings you want to delete.
+	Delete types.String `tfsdk:"delete"`
```

Comment:
I have no idea what `delete` actually does or when it would be needed.
It is also not exposed in the GUI. Should we keep it here, because it's in the API / remove it since it's not in the GUI / investigate proxmox source code to figure out how it's used?